Source/Core/Core/IOS/IOS.cpp

```diff
@@ -158,18 +155,77 @@ static bool SetupMemory(u64 ios_title_id, MemorySetupType setup_type)
   Memory::Write_U32(PLACEHOLDER, ADDR_PH3);
   Memory::Write_U32(target_imv->ios_version, ADDR_IOS_VERSION);
   Memory::Write_U32(target_imv->ios_date, ADDR_IOS_DATE);
-  Memory::Write_U32(target_imv->unknown_begin, ADDR_UNKNOWN_BEGIN);
-  Memory::Write_U32(target_imv->unknown_end, ADDR_UNKNOWN_END);
+  Memory::Write_U32(target_imv->ios_reserved_begin, ADDR_IOS_RESERVED_BEGIN);
+  Memory::Write_U32(target_imv->ios_reserved_end, ADDR_IOS_RESERVED_END);
   Memory::Write_U32(PLACEHOLDER, ADDR_PH4);
   Memory::Write_U32(PLACEHOLDER, ADDR_PH5);
   Memory::Write_U32(target_imv->ram_vendor, ADDR_RAM_VENDOR);
   Memory::Write_U8(0xDE, ADDR_BOOT_FLAG);
   Memory::Write_U8(0xAD, ADDR_APPLOADER_FLAG);
   Memory::Write_U16(0xBEEF, ADDR_DEVKIT_BOOT_PROGRAM_VERSION);
   Memory::Write_U32(target_imv->sysmenu_sync, ADDR_SYSMENU_SYNC);
+
+  RamOverrideForIOSMemoryValues(setup_type);
+
   return true;
 }
 
+// We're gonna make our own IOS Memory Values.
+void RamOverrideForIOSMemoryValues(MemorySetupType setup_type)
+{
+  if (Config::Get(Config::MAIN_RAM_OVERRIDE_ENABLE))
+  {
+    // Some unstated constants that can be inferred.
+    u32 ipc_buffer_size;
+    u32 ios_reserved_size;
+
+    {
+      const u32 ipc_buffer_begin = Memory::Read_U32(ADDR_IPC_BUFFER_BEGIN);
+      const u32 ipc_buffer_end = Memory::Read_U32(ADDR_IPC_BUFFER_END);
+      const u32 ios_reserved_begin = Memory::Read_U32(ADDR_IOS_RESERVED_BEGIN);
+      const u32 ios_reserved_end = Memory::Read_U32(ADDR_IOS_RESERVED_END);
+
+      // Inferring them
+      ipc_buffer_size = ipc_buffer_end - ipc_buffer_begin;
```

Comment:
Why not just:

```cpp
const u32 ipc_buffer_size = ipc_buffer_end - ipc_buffer_begin;
```

or even 

```cpp
const u32 ipc_buffer_size =
  Memory::Read_U32(ADDR_IPC_BUFFER_END) - Memory::Read_U32(ADDR_IPC_BUFFER_BEGIN);
```

I don't think it's worth introducing a new scope *and* making some variables non-const.